test/io/test_io.py

```diff
@@ -993,6 +993,37 @@ def sleep(i=i):
         assert delta > 1 and delta < 1.5
 
 
+def test_pool_acquisition_timeout(defaultenv, metapostgrest):
+    "Verify that PGRST_DB_POOL_ACQUISITON_TIMEOUT times out when the pool is empty"
+
+    env = {
+        **defaultenv,
+        "PGRST_DB_POOL": "1",
+        "PGRST_DB_POOL_ACQUISITION_TIMEOUT": "1",  # 1 second
+    }
+
+    with run(env=env) as postgrest:
```

Comment:
run has a `no_pool_connection_available` parameter, could that be used for this test instead of opening a thread?

Mostly to be consistent with other tests already using that.

Reply:
Unfortunately, that parameter doesn't actually do anything, compare #2456 #2457. It's a good point that we should be using this though. It shouldn't be too hard to fix it, using a similar approach as here.

Reply:
I've rebased this on top of #2457 (which now has a fix for the issue), should work now!